lxd/network/driver_ovn.go

```diff
@@ -5442,8 +5442,30 @@ func (n *ovn) Leases(projectName string, clientType request.ClientType) ([]api.N
 	var err error
 	leases := []api.NetworkLease{}
 
-	// If requested project matches network's project then include gateway IPs.
-	if projectName == n.project || projectName == "" {
+	var project *api.Project
+
+	// Only query for project when we need to check for "features.networks". That is, when the network project is
+	// the default project and projectName refers to any one other project.
+	if projectName != api.ProjectDefaultName && projectName != "" && n.project == api.ProjectDefaultName {
+		err := n.state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+			dbProject, err := dbCluster.GetProject(ctx, tx.Tx(), projectName)
+			if err != nil {
+				return err
+			}
+
+			project, err = dbProject.ToAPI(ctx, tx.Tx())
+
+			return err
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Include gateway IPs when the network belongs to requested project, getting leases for all projects or
+	// requested project doesn't have features.networks and network is in default project.
+	if projectName == n.project || projectName == "" ||
```

Comment:
please dont put if statements on multiple lines.
If its too long please break it up into multiple variables to make the if statement clearer.